src/plugins/telemetry/public/services/telemetry_notifications/telemetry_notifications.test.ts

```diff
@@ -44,25 +26,38 @@ describe('setOptedInNoticeSeen', () => {
 });
 
 describe('shouldShowOptedInNoticeBanner', () => {
-  it("should return true because a banner hasn't been shown, the notice hasn't been seen and the user has privileges to edit saved objects", () => {
+  describe(`when the banner isn't visible yet`, () => {
     const telemetryService = mockTelemetryService();
-    telemetryService.getUserShouldSeeOptInNotice = jest.fn().mockReturnValue(true);
+    const getUserShouldSeeOptInNotice = jest.fn();
+    telemetryService.getUserShouldSeeOptInNotice = getUserShouldSeeOptInNotice;
     const telemetryNotifications = mockTelemetryNotifications({ telemetryService });
-    expect(telemetryNotifications.shouldShowOptedInNoticeBanner()).toBe(true);
-  });
 
-  it('should return false because the banner is already on screen', () => {
-    const telemetryService = mockTelemetryService();
-    telemetryService.getUserShouldSeeOptInNotice = jest.fn().mockReturnValue(true);
-    const telemetryNotifications = mockTelemetryNotifications({ telemetryService });
-    telemetryNotifications['optedInNoticeBannerId'] = 'bruce-banner';
-    expect(telemetryNotifications.shouldShowOptedInNoticeBanner()).toBe(false);
+    it('should return true when `telemetryService.getUserShouldSeeOptInNotice returns true', () => {
+      getUserShouldSeeOptInNotice.mockReturnValue(true);
+      expect(telemetryNotifications.shouldShowOptInStatusNoticeBanner()).toBe(true);
+    });
+
+    it('should return false when `telemetryService.getUserShouldSeeOptInNotice returns false', () => {
+      getUserShouldSeeOptInNotice.mockReturnValue(false);
+      expect(telemetryNotifications.shouldShowOptInStatusNoticeBanner()).toBe(false);
+    });
   });
 
-  it("should return false because the banner has already been seen or the user doesn't have privileges to change saved objects", () => {
+  describe(`when the banner is already visible`, () => {
     const telemetryService = mockTelemetryService();
-    telemetryService.getUserShouldSeeOptInNotice = jest.fn().mockReturnValue(false);
+    const getUserShouldSeeOptInNotice = jest.fn();
+    telemetryService.getUserShouldSeeOptInNotice = getUserShouldSeeOptInNotice;
     const telemetryNotifications = mockTelemetryNotifications({ telemetryService });
-    expect(telemetryNotifications.shouldShowOptedInNoticeBanner()).toBe(false);
+    telemetryNotifications['optInStatusNoticeBannerId'] = 'bruce-banner';
+
+    it('should return false when `telemetryService.getUserShouldSeeOptInNotice returns true', () => {
```

Comment:
nitnit, same for below
```suggestion
    it('should return false when `telemetryService.getUserShouldSeeOptInNotice` returns true', () => {
```